scripts/test/generate_lld_tests.py

```diff
@@ -66,8 +66,8 @@ def generate_wast_files(llvm_bin, emscripten_root):
             '-z', '-stack-size=1048576',
             obj_path, '-o', wasm_path,
             '--allow-undefined',
+            '--strip-debug',
```

Comment:
why strip debug info? I think that's why function names are lost in the new outputs, making it harder to read them (and to compare to before, to understand the patch)

Reply:
Because this bug occurred because we were inadvertently relying on debug names in this pass.

None of the stuff we do in wasm-emscripten-finalize should depend on the debug name section (I think).

Reply:
Hopefully they should become more readable again if we implement #2504... but that might least to the same kind of sematic bugs I guess... e.g. using a functions name when you meant to refer to its import name

Reply:
I don't know wasm-emscripten-finalize well enough to know if it depends on function names from the name section or not. cc @jgravelle-google

Reply:
In emscripten `-O2` and above we pass `--strip-debug` to lld which means the binary that wasm-emscripten-finalize sees in those cases doesn't contain any debug name section.   This is why this particular bug only shows up at `-O2`.

What is more I think its important that continue to *not* depend on debug names so I think both part of this change are warranted.